en/concept/flight_tasks.md

```diff
@@ -2,8 +2,141 @@
+
+The instructions below might be used to create a task named *MyTask*:
+
+1. Create a directory for the new flight task in [PX4-Autopilot/src/lib/flight_tasks/tasks/](https://github.com/PX4/PX4-Autopilot/tree/master/src/lib/flight_tasks/tasks).
+   By convention the directory is named after the task, so we will call it **/MyTask**.
+   ```
+   mkdir PX4-Autopilot/src/lib/flight_tasks/tasks/MyTask
+   ```
+2. Create empty source code and *cmake* files for the new flight task in the *MyTask* directory using the prefix "FlightTask":
+   - CMakeLists.txt
+   - FlightTaskMyTask.hpp
+   - FlightTaskMyTask.cpp
+3. Update **CMakeLists.txt** for the new task
+   - Copy the contents of the **CMakeLists.txt** for another task - e.g. [AutoMapper/CMakeLists.txt](https://github.com/PX4/PX4-Autopilot/blob/master/src/lib/flight_tasks/tasks/AutoMapper/CMakeLists.txt)
+   - Update the copyright to the current year
+     ```cmake   
+     ############################################################################
+     #
+     #   Copyright (c) 2021 PX4 Development Team. All rights reserved.
+     #
+     ```
+   - Modify the code to reflect the new task - e.g. replace `FlightTaskAutoMapper` with `FlightTaskMyTask`.
+     The code will look something like this:
+     ```cmake 
+     px4_add_library(FlightTaskMyTask
+         FlightTaskMyTask.cpp
+     )
+
+     target_link_libraries(FlightTaskMyTask PUBLIC FlightTask)
+     target_include_directories(FlightTaskMyTask PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+     ```
+
+4. Update the header file (in this case **FlightTaskMyTask.hpp**):
+   Most tasks reimplement the virtual methods `activate()` and `update()`, and in this example we also have a private variable.
+   ```cpp
+   #pragma once
+
+   #include "FlightTask.hpp"
+
+   class FlightTaskMyTask : public FlightTask
+   {
+   public:
+     FlightTaskMyTask() = default;
+     virtual ~FlightTaskMyTask() = default;
+
+     bool update();
+     bool activate(vehicle_local_position_setpoint_s last_setpoint);
+
+   private:
+     float _origin_z = 0.0f;
+   };
+   ```
+4. Update the cpp file as appropriate.
+   This example provides as simple implementation of **FlightTaskMyTask.cpp** that simply indicates that the task methods are called.
+   ```cpp
+   #include "FlightTaskMyTask.hpp"
+
+   bool FlightTaskMyTask::activate(vehicle_local_position_setpoint_s last_setpoint)
+   {
+     bool ret = FlightTask::activate(last_setpoint);
+     PX4_INFO("FlightTaskMyTask activate was called! ret:%d", ret); // check if engaged this mode.
+     return ret;
+   }
+
+   bool FlightTaskMyTask::update()
+   {
+     PX4_INFO("FlightTaskMyTask update was called!"); // check updated.
+     return true;
+   }
+   ```
+5. Add the new task to the list of tasks to be build in [/src/lib/flight_tasks/tasks/CMakeLists.txt](https://github.com/PX4/PX4-Autopilot/blob/master/src/lib/flight_tasks/tasks/CMakeLists.txt):  
```

Comment:
@yutaide @MaEtUgR This does not appear to be true for master - I THINK you don't need to do this because CMAKE just searches the current tree for file names. Correct? Or do you explicitly have to add them still?

Reply:
@hamishwillee 
Thank you for your rework!
I'm not sure if this step is necessary or not since I'm not familiar with CMAKE.
@MaEtUgR  What do you think about this?

Reply:
Thank you so much @hamishwillee !
You still have to add the name of your task but it's here on master:
https://github.com/PX4/PX4-Autopilot/blob/master/src/modules/flight_mode_manager/CMakeLists.txt#L40